opentelemetry-sdk/src/opentelemetry/sdk/trace/propagation/b3_format.py

```diff
@@ -95,19 +97,33 @@ def extract(
                 or flags
             )
 
+        if (
+            fullmatch(r"[\da-fA-F]{16}|[\da-fA-F]{32}", trace_id) is None or
+            fullmatch(r"[\da-fA-F]{16}", span_id) is None
+        ):
+            trace_id = generate_trace_id()
+            span_id = generate_span_id()
```

Comment:
So I think the entire header is thrown out, so the sampled and flags headers will just do their default behavior as if they were never passed through.

Reply:
Fixed